client/my-sites/plans/index.js

```diff
@@ -73,6 +74,11 @@ export default function () {
 		p2RedirectToHubPlans,
 		currentPlan
 	);
+	trackedPage(
+		'/plans/my-plan/trial-upgrade-confirmation/:domain',
```

Comment:
In terms of the path, I think this matches our thinking, but not the user's. Maybe simplifying this to `my-plan/trial-upgraded/:domain` would be enough? That way the path reflects what just happened, rather than what the page is.